### ISSUE TITLE
Possible to have sensor for Sound Mode change?

### ISSUE THREAD

**Issue by SawkeeReemo** [grzegorz914/homebridge-denon-tv#253]
Is it possible, or is there a way to have a sensor that triggers when a Sound Mode changes?

(sorry, not an issue, but I didn’t see a place to ask a general question. Hope this is ok.)

**Comment by grzegorz914** [comment]
Already there
![IMG_0949](https://github.com/grzegorz914/homebridge-denon-tv/assets/28867813/0c026e49-4fc2-4918-ae39-543012d6accb)

**Comment by SawkeeReemo** [comment]
Oh, I wa confused because it says it’s trigger on input change instead of mode change. Definitely not changing inputs. I’ll give it a try, thank you!

**Comment by SawkeeReemo** [comment]
Sorry to open this back up, but after several days I cannot figure out how to use this to detect what sound mode I switch to. Is there a way to do that?

I tried “when sound mode sensor opens” but then I don’t know how to make an automation that says “if Sound Mode is AUTO, then” do these actions. I tried “name contains/is” and “configured name contains/is,” but neither seemed to trigger. 

Any tips? And thank you in advance. 
![IMG_8100](https://github.com/grzegorz914/homebridge-denon-tv/assets/60272670/4ef736df-19ab-4823-9f4e-b1b8d9b9c944)

**Comment by grzegorz914** [comment]
To do this You can create custom sensor with reference AUTO and display type Occupancy or Motion Sensor, in this case the sensor fired if switch to this Surround Mode

**Comment by grzegorz914** [comment]
Sound Mode AUTO doesn’t exist.

**Comment by grzegorz914** [comment]
Ups, right now sound mode are not available for custom sensors, will add it soon

**Comment by SawkeeReemo** [comment]
Oh that’d be great! Thank you!

**Comment by grzegorz914** [comment]
added in 4.11.0

**Comment by grzegorz914** [comment]
Yes You can set AVR to AUTO but the receiver switch between surrounds mode based on audio meta data and report for example Dolby Digital+ or Dolby Surround, etc...

**Comment by SawkeeReemo** [comment]
Oh yeah, I see what you mean. It stay in "AUTO" for the mode, but the receiver is showing the actual codec. I'll have to play around. Thanks again for this!
